app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -61,18 +62,27 @@ def collect_disks
     disks = hardware.disks.map { |disk| "#{disk.storage.ems_ref}/disks/#{disk.filename}" }
     vm_disks = []
 
-    ext_management_system.with_provider_connection do |rhevm|
-      disks.each do |disk|
-        begin
-          vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
-        rescue Ovirt::MissingResourceError
-          nil
+    unless ext_management_system.nil?
+      ext_management_system.with_provider_connection do |rhevm|
+        disks.each do |disk|
+          begin
+            vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
+          rescue Ovirt::MissingResourceError
+            nil
+          end
         end
       end
     end
+
     vm_disks
   end
 
+  def disconnect_inv
+    disconnect_storage
```

Comment:
If this method is called twice I think it will always say that the storage was disconnected the second time because you skip collecting the disks and always return `vm_disks = []`
Maybe we should leave `collect_disks` the way it is but only call `disconnect_storage` if it isn't already archived?